types/wlr_keyboard.c

```diff
@@ -41,8 +41,6 @@ static void keyboard_modifier_update(struct wlr_keyboard *keyboard) {
 	keyboard->modifiers.latched = latched;
 	keyboard->modifiers.locked = locked;
 	keyboard->modifiers.group = group;
-
-	wl_signal_emit(&keyboard->events.modifiers, keyboard);
```

Comment:
Why did you move the `emit` call outside this function? The X11 backend is still not calling `wlr_keyboard_notify_modifiers` so emiting the event here is necessary. Plus, the purpose of this function is to only emit the event when needed (when modifiers really changed).

Reply:
I still believe we should have the event, but I'm going to undo the change here to clean up the pr and discuss it as a separate issue.